resident/resident-service/src/main/java/io/mosip/resident/service/impl/ResidentCredentialServiceImpl.java

```diff
@@ -0,0 +1,264 @@
+							PartnerResponseDto.class);
+					additionalAttributes.put("partnerName",
+							partnerResponseDto.getOrganizationName());
+					additionalAttributes.put("encryptionKey", credentialReqestDto.getEncryptionKey());
+					sendNotification(dto.getIndividualId(), NotificationTemplateCode.RS_CRE_REQ_SUCCESS, null);
+			   } else {
+				logger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.APPLICATIONID.toString(),
+						LoggerFileConstant.APPLICATIONID.toString(),
+						ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorMessage());
+				sendNotification(dto.getIndividualId(), NotificationTemplateCode.RS_CRE_REQ_FAILURE, null);
+				throw new ResidentServiceException(ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorCode(),
+						ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorMessage());
+			}
+		}/* catch (OtpValidationFailedException e) {
+			sendNotification(dto.getIndividualId(), IdType.valueOf("UIN"),
+					NotificationTemplateCode.RS_CREDENTIAL_FAILURE, null);
+			throw new ResidentServiceException(ResidentErrorCode.OTP_VALIDATION_FAILED.getErrorCode(),
+					e.getErrorText(), e);
+		}*/
+		catch (ResidentServiceCheckedException e) {
+			sendNotification(dto.getIndividualId(), NotificationTemplateCode.RS_CRE_REQ_FAILURE, null);
+			throw new ResidentCredentialServiceException(ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorMessage(), e);
+		}
+		catch (ApisResourceAccessException e) {
+			sendNotification(dto.getIndividualId(), NotificationTemplateCode.RS_CRE_REQ_FAILURE, null);
+			throw new ResidentCredentialServiceException(ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.API_RESOURCE_ACCESS_EXCEPTION.getErrorMessage(), e);
+		}
+		catch (IOException e) {
+			sendNotification(dto.getIndividualId(), NotificationTemplateCode.RS_CRE_REQ_FAILURE, null);
+			throw new ResidentCredentialServiceException(ResidentErrorCode.IO_EXCEPTION.getErrorCode(),
+					ResidentErrorCode.IO_EXCEPTION.getErrorMessage(), e);
+		}
+
+		return residentCredentialResponseDto;
+	}
+	
+	@Override
+	public CredentialRequestStatusResponseDto getStatus(String requestId) {
+		ResponseWrapper<CredentialRequestStatusResponseDto> responseDto = null;
+		CredentialRequestStatusResponseDto credentialRequestStatusResponseDto=new CredentialRequestStatusResponseDto();
+		try {
+			String credentialUrl=credentailStatusUrl+requestId;
+			URI credentailStatusUri = URI.create(credentialUrl);
+			responseDto =residentServiceRestClient.getApi(credentailStatusUri, ResponseWrapper.class, tokenGenerator.getToken());
+			credentialRequestStatusResponseDto = JsonUtil
+						.readValue(JsonUtil.writeValueAsString(responseDto.getResponse()), CredentialRequestStatusResponseDto.class);
+
+			sendNotification(credentialRequestStatusResponseDto.getId(), NotificationTemplateCode.RS_CRE_STATUS,
```

Comment:
Please check in status template,statuscode and request id is present ,please add in additional attributes